src/main/java/org/apache/sling/event/impl/support/PropertyNameAndOperationExtractor.java

```diff
@@ -0,0 +1,37 @@
+package org.apache.sling.event.impl.support;
+
+public class PropertyNameAndOperationExtractor {
+    public PropertyNameAndOperation extractPropertyNameAndOperation(String key) {
```

Comment:
I would convert this into a static method, then you won't need to invoke the constructor anymore.